src/components/FileUploaderDetails/Example.md

```diff
@@ -46,14 +47,28 @@ class FileUploaderDetailsExample extends React.Component {
         if (file.progress < 100) {
           setTimeout(() => this.makeProgress(id), timeout);
         } else {
-          this.eventManager.emit(FileUploadActions.uploadSuccess, { files: [file] });
+          if (!this.errorShown) {
+            this.errorShown = true;
+            this.eventManager.emit(FileUploadActions.uploadFailure, { files: [{...file, error: 'upload_204'}] });
+          } else {
+            this.eventManager.emit(FileUploadActions.uploadSuccess, { files: [file] });
+          }
         }
       }
 
       this.files[file.fileId] = file;
     }
   };
 
+  translate(error) {
+    switch(error) {
+      case 'upload_204':
+        return 'Localized message for error 204';
```

Comment:
What error is error 204? It seems like this has nothing to do with precise ui. Maybe model error not as a string, but rather as a function of the error? Then in the standard implementation (this one) we could make this switch, otherwise it would be left to the user to write something meaningful based on an error object.

Reply:
this is just an example, there could be any text

Reply:
As Maksym mentioned, it's an example, but it's actually the example based on the issue we have currently. Currently when the error is emitted from the consumer application, the status table just displays the error, and then the consumer has no chance of localizing it. Like in the example above, the service can throw different types of error, and we could use this new function to localize the errors based on e.g. error code returned from the service.

Reply:
> As Maksym mentioned, it's an example, but it's actually the example based on the issue we have currently.

This issue can be also solved by adding an `error` function to the `labels` (passing in the exact error / error object). It also is consistent with the current approach.

Reply:
Approach with `setLabels` have a big problem right now - it doesn't allow to define labels per component. We can introduce `error` function that we will need `notification` and so on,  or we can have a standard function responsible for all translations

I doubt that precise-ui user will be using all components, which means that the scenario where he needs to provide all texts through `setLabel` is much more rare than he need to provide texts to the concrete component. 

The concrete component will have then `setTexts` prop with a type 
```
interface SetTexts {
  (str: 'openLabel' | 'closeLabel'): string
}
```

Reply:
No I think you are wrong. The current approach allows both.

Please check the use right now - you have the globals, which *can* be set using *setLabels* and are used as *fallbacks* when the labels are *not set* on the component.

> I doubt that precise-ui user will be using all components, which means that the scenario where he needs to provide all texts through `setLabel` is much more rare than he need to provide texts to the concrete component.

I know at least one user / use that *uses* all components. And I think you also know this application ... so I wonder why you think there is no such case?

Reply:
`setLabels` for providing globals and fallbacks still can be kept with only extending type by `{[nonDefinedLabel: string]: string}`. this solves one problem

But there is second one - ### texts per component. In fact we already have a problem with this: texts to `FileUploader` currently can be provided globally only through the `setLabels`. There is at least 16 labels related to `FileUploader`. With current approach if I would like to customize `FileUploader` I will need to introduce 16 new props for labels which is just insane.

If we have setLabels globally why just not to have setLabels exclusively  for the component as well?